base/uk.ac.stfc.isis.ibex.scriptgenerator/src/uk/ac/stfc/isis/ibex/scriptgenerator/ScriptGeneratorSingleton.java

```diff
@@ -390,4 +397,69 @@ public void refreshParameterValidityChecking() {
 			threadError = true;
 		}
 	}
+	
+	/**
+	 * Generate a script and save it to file.
+	 * 
+	 * @return The filepath the script is saved to.
+	 * @throws InvalidParamsException If the parameters are invalid a script cannot be generated.
+	 * @throws IOException If we fail to create and write to a file.
+	 */
+	public String generate() throws InvalidParamsException, IOException {
+		return generateTo("C:/Scripts/");
+	}
+	
+	/**
+	 * Generate a script and save it to file with the given prefix.
+	 * 
+	 * @return The filepath the script is saved to.
+	 * @throws InvalidParamsException If the parameters are invalid a script cannot be generated.
+	 * @throws IOException If we fail to create and write to a file.
+	 */
+	public String generateTo(String filepathPrefix) throws InvalidParamsException, IOException {
+		// Generate the script
+		String[] generatedScript = generateScript();
+		// Create the filename and file
+		String timestamp = getTimestamp();
+		int version = 0;
+		String filepath;
+		File file;
+		do {
+			if (version == 0) {
+				filepath = String.format("%s%s-%s.py", filepathPrefix, getConfig().getName(), timestamp);
+			} else {
+				filepath = String.format("%s%s-%s(%s).py", filepathPrefix, getConfig().getName(), timestamp, version);
+			}
+			file = new File(filepath);
+			file.getParentFile().mkdirs();
+			version += 1;
+		} while(!file.createNewFile());
```

Comment:
This method is doing too much:
- Formatting a filename
- Creating a file
- Writing the script to it

Can we split this up into 2 or 3 appropriately named methods to make it easier to understand.